src/openms/source/ANALYSIS/TOPDOWN/FLASHDeconvAlgorithm.cpp

```diff
@@ -0,0 +1,2109 @@
+    }
+    return deconvolved_spectrum_;
+  }
+
+  void FLASHDeconvAlgorithm::updateMembers_()
+  {
+    min_mz_ = param_.getValue("min_mz");
+    max_mz_ = param_.getValue("max_mz");
+
+    min_rt_ = param_.getValue("min_rt");
+    max_rt_ = param_.getValue("max_rt");
+
+    min_abs_charge_ = param_.getValue("min_charge");
+    max_abs_charge_ = param_.getValue("max_charge");
+    is_positive_ = min_abs_charge_ > 0;
+
+    min_abs_charge_ = abs(min_abs_charge_);
+    max_abs_charge_ = abs(max_abs_charge_);
+
+    if (min_abs_charge_ > max_abs_charge_)
+    {
+      int tmp = min_abs_charge_;
+      min_abs_charge_ = max_abs_charge_;
+      max_abs_charge_ = tmp;
+    }
+
+    max_mass_ = param_.getValue("max_mass");
+    min_mass_ = param_.getValue("min_mass");
+
+    isolation_window_size_ = param_.getValue("isolation_window");
+
+    intensity_threshold_ = param_.getValue("min_intensity");
+    min_support_peak_count_ = param_.getValue("min_peaks");
+
+    bin_width_.clear();
+    tolerance_ = param_.getValue("tol");
+
+    for (double& j : tolerance_)
+    {
+      j *= 1e-6;
+      bin_width_.push_back(.5 / j);
+    }
+
+    min_isotope_cosine_ = param_.getValue("min_isotope_cosine");
+    max_mass_count_ = param_.getValue("max_mass_count");
+    rt_window_ = param_.getValue("rt_window");
+  }
+
+  FLASHDeconvHelperStructs::PrecalculatedAveragine FLASHDeconvAlgorithm::getAveragine()
+  {
+    return avg_;
+  }
+
+  void FLASHDeconvAlgorithm::calculateAveragine(const bool use_RNA_averagine)
+  {
+    auto generator = new CoarseIsotopePatternGenerator();
+
+    auto iso = use_RNA_averagine ?
+                   generator->estimateFromRNAWeight(max_mass_) :
+                   generator->estimateFromPeptideWeight(max_mass_);
+    iso.trimRight(0.01 * iso.getMostAbundant().getIntensity());
+
+    generator->setMaxIsotope(iso.size());
+    avg_ = FLASHDeconvHelperStructs::PrecalculatedAveragine(50, max_mass_, 25, generator, use_RNA_averagine);
+    avg_.setMaxIsotopeIndex(iso.size() - 1);
+  }
+
+  // generate filters
+  void FLASHDeconvAlgorithm::setFilters_()
+  {
+    filter_.clear();
+    harmonic_filter_matrix_.clear();
+    int charge_range = current_max_charge_ - current_min_charge_ + 1;
+    for (int i = 0; i < charge_range; i++)
+    {
+      filter_.push_back(log(1.0 / (i + current_min_charge_)));//+
+    }
+
+    harmonic_filter_matrix_.resize(harmonic_charges_.size(), charge_range);
+
+    for (Size k = 0; k < harmonic_charges_.size(); k++)
+    {
+      int hc = harmonic_charges_[k];
+      int n = hc / 2;
+
+      for (int i = 0; i < charge_range; i++)
+      {
+        harmonic_filter_matrix_
+            .setValue(k, i, log(1.0 / (-1.0 * n / hc + (i + current_min_charge_))));// + current_min_charge_
+      }
+    }
+  }
+
+  //Generate uncharged log mz transformated peaks
+  void FLASHDeconvAlgorithm::updateLogMzPeaks_(const MSSpectrum* spec)
+  {
+    std::vector<LogMzPeak>().swap(log_mz_peaks_);
```

Comment:
```suggestion
    log_mz_peaks_.clear();
```

Reply:
I used this swap trick to completely erase vector<>.  When clear() was used in the place of swap(), FLASHDeconv suffered from memory issue at least in mac osx. I will double check if it is still the case.

Reply:
mhh.. LogMzPeak is not that huge of a struct.
This vector needs to be HUGE in order for this to make a difference.
Also, you are repopulating it down below again (`log_mz_peaks_.reserve(spec->size());`),
so if the spectra are of comparable size, there should be no win in swapping it out...